src/ddscxx/include/org/eclipse/cyclonedds/domain/qos/DomainParticipantQosDelegate.hpp

```diff
@@ -56,11 +53,13 @@ class OMG_DDS_API DomainParticipantQosDelegate
     void check() const;
 
     bool operator ==(const DomainParticipantQosDelegate& other) const;
-    DomainParticipantQosDelegate& operator =(const DomainParticipantQosDelegate& other);
 
+    const uint64_t &present() const {return present_;}
+    uint64_t &present() {return present_;}
 private:
-    dds::core::policy::UserData user_data_;
-    dds::core::policy::EntityFactory entity_factory_;
+    uint64_t                          present_ = 0;
+    dds::core::policy::UserData       user_data_;
+    dds::core::policy::EntityFactory  entity_factory_;
```

Comment:
I tried to find out whether the default constructor for the various policies will initialise the value properly, but I am not quite sure that it really does: for example, `EntityFactory` seems to end up in `PolicyDelegate.hpp:222` with the default-default constructor, and IINM, that means the boolean in there won't be initialized. (Or perhaps that was for POD and classes are different?)

I don't think there have been any changes to this aspect, but I think it is worth double-checking.

Reply:
I will double check this.

Reply:
The old iteration with the explicit default constructor also takes the default constructor for the EntityFactory QoSPolicy.
Maybe fix it at the default constructor for the QoSPolicy?